subsys/bluetooth/services/bas_c.c

```diff
@@ -272,3 +388,40 @@ int bt_gatt_bas_c_get(struct bt_gatt_bas_c *bas_c)
 	}
 	return bas_c->battery_level;
 }
+
+
+int bt_gatt_bas_c_periodic_read_start(struct bt_gatt_bas_c *bas_c,
+				      s32_t interval,
+				      bt_gatt_bas_c_read_cb func)
+{
+	if (!bas_c || !func) {
+		return -EINVAL;
+	}
+
+	if (bt_gatt_bas_c_notify_supported(bas_c)) {
+		return -ENOTSUP;
+	}
+
+	bas_c->periodic_read.read_cb = func;
+	atomic_set(&bas_c->periodic_read.interval, interval);
+
+	if (!atomic_test_and_set_bit(&bas_c->periodic_read.process,
+				     BAS_PERIODIC_READ_PROCESS)) {
+		k_delayed_work_submit(&bas_c->periodic_read.read_work, interval);
+	}
+
+	return 0;
+}
+
+
+void bt_gatt_bas_c_periodic_read_stop(struct bt_gatt_bas_c *bas_c)
+{
+	/* If delayed workqueue pending, cancel it. */
+	k_delayed_work_cancel(&bas_c->periodic_read.read_work);
```

Comment:
You should probably check that it is actually running here before canceling it. Since it is undefined behavior if the work is not yet submitted.

Reply:
Commented in previous conversation